### DISCUSSION TITLE
Uploading images results in file permission 600

### DISCUSSION THREAD

**Discussion by hedho** [datenstrom/community#333] [Report a bug] [answered]
Doesn't show the images because it changes file permission to 600, I am using https://github.com/datenstrom/yellow-developers

**Comment by schulle4u** [comment]
Confirmed. What is your exact LAMPP setup? I just tested on two different systems - one running php in fpm mode, the other with just the builtin development webserver. As it seems file chmod 600 only has a negative effect on systems running php-fpm. I currently don't have access to a machine with php as apache module to test it.

**Comment by hedho** [comment]
I am using Debian with Plesk Nginx/Apache fpm and PHP both of them!

**Comment by schulle4u** [comment]
Did some research. I'm not fully into yellow's core functions, but if I'm right Yellow uses a toolbox function to process uploaded files, that first creates a temporary file and then uses php's `rename()` function to provide the actual uploaded file. During this process, the file permissions are set to 0600. As it seems that this is a [known problem in php](https://stackoverflow.com/questions/15055391/uploading-files-with-php-results-in-permissions-600). A workaround is to use `copy()` instead of `rename()`, or simply do an additional `chmod()` on the uploaded files after renaming them.

**Comment by hedho** [comment]
Issue Fixed ! and CLOSED ! - renamefile + copyfile

**Comment by schulle4u** [comment]
We should leave this issue opened to let main developers know that a permanent fix is needed. (CC @markseu)

**Comment by markseuffert** [comment]
@schulle4u Would be interested to investigate. Can you give me access to a server with php-fpm?

**Comment by schulle4u** [comment]
No problem, I'll pm you with further details.

**Comment by markseuffert** [comment, marked as answer]
Thank you Steffen. Uploaded files and created files did not have the same permissions. This is why it didn't show the uploaded images on some web servers. I changed the editor so that all files get the same permissions, which is 644 per default.

![screen shot 2018-06-21 at 15 07 57](https://user-images.githubusercontent.com/2429165/41722877-f8bc626a-7569-11e8-98fa-67f31d371779.png)

Should be working now.

**Comment by BsNoSi** [comment]
Had this, too. Tested with success, the images show up as espected. → simply updated the edit plugin.
